packages/models/src/sql/log.js

```diff
@@ -0,0 +1,29 @@
+import { open, write } from 'fs';
+
+let SqlWarningFileName = 'sql_warning.txt';
+let messages /** @type {string[]} */ = [];
+let writeMessage = (msg) => messages.push(msg);
+
+process.on('exit', () => {
```

Comment:
Why not print messages as they come?

Reply:
Well, this is what we were doing before, and it was very messy with all the messages getting printed to the console and inter-mingling with the console output. In any case, exit hook only takes effect if we couldn't open the sql_warning.txt file.

Reply:
I think it's preferable to print them as they come (even if it's messy), but if not, then I'd add some header to say what the dump is about and that it's been collected over the whole run. Otherwise it can be confusing if you don't already know what it is.